### ISSUE TITLE
Nestrov's accelerated gradient

### ISSUE THREAD

**Issue by sanfernoronha** [RyanDsilva/nn-from-scratch#6]
Another optimizer

**Comment by RyanDsilva** [comment]
Let's probably do the roadmap functions first? RMSProp and Adam are pretty popular optimizers

**Comment by sanfernoronha** [comment]
Yeah but the thing is nestrov is a better implementation of momentum and i guess implementing it will reshape how we write the code for all our optimizers and make it more general

**Comment by sanfernoronha** [comment]
Let's add nestrov to the road map itself!

**Comment by RyanDsilva** [comment]
Okay, I'll change the roadmap a little bit today to include everything in categories

**Comment by sanfernoronha** [comment]
Implementing nestrov within the momentum class.
Way to use:

`nn.useOptimizer(Momentum(nestrov=True), learning_rate=config.learning_rate , beta = config.beta)`
:+1: 
Turns out nestrov needs to find gradient of cost function at the next approx parameter instead of current parameter(weight/bias). :no_good: 
Looking for way to manipulate around it :bulb:

**Comment by sanfernoronha** [comment]
@RyanDsilva 
:skull:  Nestrov's approximation formula needs review. The results should be slightly better than simple momentum but the results are slightly worse. 
Also, material online shows multiple implementations so I have picked the best one to work on with slight modification which only then gives correct result.
